node/modules/services.go

```diff
@@ -73,14 +76,45 @@ func RunBlockSync(h host.Host, svc *blocksync.BlockSyncService) {
 	h.SetStreamHandler(blocksync.BlockSyncProtocolID, svc.HandleStream)
 }
 
-func HandleIncomingBlocks(mctx helpers.MetricsCtx, lc fx.Lifecycle, ps *pubsub.PubSub, s *chain.Syncer, bserv dtypes.ChainBlockService, chain *store.ChainStore, stmgr *stmgr.StateManager, h host.Host, nn dtypes.NetworkName) {
-	ctx := helpers.LifecycleCtx(mctx, lc)
+func waitForSync(stmgr *stmgr.StateManager, blocks int, subscribe func()) {
+	nearsync := uint64(blocks) * uint64(build.BlockDelaySecs) * uint64(time.Second) //nolint
```

Comment:
I think this should be named tipset to be more clear as to how do we measure time.

Reply:
sure, that's easy to do.

Reply:
Or epochs

Reply:
presumably you mean the `blocks` variable, right?

Reply:
done, renamed `block` to `tipsets`.

Reply:
actually, epochs is a better name, let me do that.

Reply:
renamed yet again to epochs.